### PR TITLE
Folder reorganization: src/ layout, blueprints/, ck-projects/ unified scratch

### LINKED ISSUE #437: Folder reorganization (atomic): projects/, src/, and model_organism split
## Summary

Committed plan for a folder reorganization aligning cruijff_kit with a unified "blueprints in repo, runs on scratch" model. **Atomic PR** — one big change, internally chunked for reviewability.

_This issue body supersedes the earlier draft. Full plan doc is in `edi-ccore/reviews/folder_reorg_plan_2026-04-22.md` (private); the relevant content is summarized below for group visibility._

### Top-level repo layout (post-reorg)

```
cruijff_kit/
├── pyproject.toml               # src layout
├── src/cruijff_kit/             # all library code
│   ├── tools/
│   │   ├── model_organisms/     # was sanity_checks/model_organisms/
│   │   ├── inspect/
│   │   ├── torchtune/
│   │   ├── slurm/
│   │   └── experiment/
│   ├── utils/
│   └── text_gen/
├── projects/                    # blueprints (read-only, never executed from)
│   ├── capitalization/
│   ├── folktexts/
│   └── model_organism/
├── tests/
├── docs/
└── [README, CLAUDE.md, etc.]
```

### Blueprint canonical shape (5 slots, 3 required)

```
projects/{name}/
├── README.md          # REQUIRED
├── inspect_task.py    # REQUIRED
├── generate_data.py   # REQUIRED
├── modifiers/         # optional (project-specific data transforms)
└── baseline.py        # optional (non-LLM comparison)
```

Earlier drafts had `lib/` and `templates/` slots; both dropped as YAGNI (no current project needs them; when primitives stabilize they graduate to `src/cruijff_kit/tools/`).

### Scratch-side structure (Option 3b — per-experiment snapshot)

```
ck-projects/{project}/
├── README.md                         # user's cross-run notes
├── findings_across_runs.md           # optional user docs
├── 2026-04-25_parity_bits_r4/
│   ├── code/                         # blueprint snapshot at run time
│   ├── checkpoints/
│   ├── logs/
│   ├── eval_output/
│   └── plots/
└── ...
```

**No living `code/` at project root** — each experiment owns its own snapshot. Only two sources of truth: the repo blueprint OR a specific experiment's snapshot. Scientific reproducibility without path archaeology.

`scaffold-experiment` skill performs the blueprint copy-out, aligned with the data-generation step Sarah added in PR #414/#423 chunks 5–7.

## Naming decisions

- Library: `src/cruijff_kit/tools/model_organisms/` (plural — gallery of primitives)
- Project blueprint: `projects/model_organism/` (singular — one project picks one organism)
- Inspect task function: `@task model_organism()` (singular — one invocation, one specimen)

Consistent with biology: plural for the set, singular for a specimen.

## What gets destroyed

- **`sanity_checks/` directory** — entirely:
  - `bit_sequences/`, `count_digits/`, `majority/` → deleted (model_organism framework already covers them via rule × input × format composition)
  - `bernoulli/`, `predictable_or_not/` → deleted (not compelling / superseded)
- **Parquet support** — fully removed (~5 lines code, 8 doc/skill references, 2 test cases, 1 template). Revival cost: small PR if needed.
- **Data tier directories** (`data/red/`, `data/yellow/`, `data/green/` as tier concept) — gone; shared datasets move to `ck-data/` on scratch (already settled).
- **`experiment.type` field** — superseded by `experiment.project`.

## Schema changes to `experiment_summary.yaml`

- **Add** `experiment.project` — required, plain project name, nested under `experiment:`
- **Remove** `experiment.type` (and its `ck-sanity-checks/` vs `ck-experiments/` routing in skills)

## Model_organism library/project split

Today `sanity_checks/model_organisms/` is dual-role (library primitives + concrete inspect task). The PR splits them:

- **Library** → `src/cruijff_kit/tools/model_organisms/` (rules, formats, inputs, generate)
- **Project blueprint** → `projects/model_organism/` (README + inspect_task + thin CLI wrapper importing from the library)

Import path change: `cruijff_kit.sanity_checks.model_organisms.*` → `cruijff_kit.tools.model_organisms.*`. Affects 2 test files and `tools/experiment/prepare_data.py`.

## `src/` layout

Standard Python src layout. `pyproject.toml` gains:

```toml
[tool.setuptools.packages.find]
where = ["src"]
```

and drops the existing `[tool.setuptools.package-dir] cruijff_kit = "."` hack.

**Import paths unchanged** — users still write `from cruijff_kit.tools.X import Y`. The `src/` dir is invisible at import time; it's a build convention. Benefits: cleaner repo root, test isolation from in-repo package code, visual separation of library from projects/tests/docs.

## Pre-PR housekeeping

Eviction list:
- `experiments/folktexts/create_balanced_4223145.out` (187KB committed SLURM log)
- `experiments/capitalization/{misc,logs,templates}/`, `package-lock.json`
- `experiments/capitalization/input/words_alpha.txt` → `ck-data/`
- `experiments/capitalization/utils/*` → promote generic helpers to `src/cruijff_kit/tools/` or delete if stale
- Capitalization `README.md` — rewrite the "copy from templates/finetuning/" section to match how folktexts/model_organism describe FT config setup (user generates via `design-experiment` skill)

## In-flight exploratory work

Variable-order PublicCoverage, majority classification blog candidate, state-level ACS, 6-bit 8B follow-up — all **archived** via `archive-experiment` skill before/during the PR. No compatibility requirement for legacy paths after the reorg.

## Suggested PR chunk sequence (all atomic, for reviewability)

1. `make clean` target (✅ landed as `7c2e738`)
2. Archive in-flight experiments (may happen outside branch)
3. Eviction pass (parquet + housekeeping deletions)
4. `src/` layout migration
5. model_organism namespace move + import path updates
6. `projects/` top-level + blueprint construction (capitalization, folktexts, model_organism)
7. `sanity_checks/` destruction
8. Schema changes (`experiment.project` required, `experiment.type` removed) + skill updates
9. Docs pass (ARCHITECTURE.md, ARTIFACT_LOCATIONS.md, CLAUDE.md, PR_CHECKLIST.md, READMEs)

## Acceptance criteria

- [ ] `src/cruijff_kit/` layout adopted; `pyproject.toml` updated; all existing import paths still resolve
- [ ] `projects/` directory created with capitalization, folktexts, and model_organism blueprints (5-slot shape)
- [ ] `sanity_checks/` directory fully removed (all 5 remaining members destroyed/absorbed)
- [ ] model_organism library at `src/cruijff_kit/tools/model_organisms/`; project blueprint at `projects/model_organism/`
- [ ] Parquet support removed (code, tests, docs, template)
- [ ] `experiment.project` added as required field; `experiment.type` removed everywhere
- [ ] Skills (`design-experiment`, `scaffold-experiment`, `run-experiment`, `summarize-experiment`, `analyze-experiment`, `archive-experiment`) operate on `projects/` exclusively
- [ ] `scaffold-experiment` performs blueprint copy-out into experiment `code/` snapshot dir
- [ ] Workflow tests (`.claude/workflow_test*.yaml`) exercise the new structure and pass
- [ ] Documentation updated: ARCHITECTURE.md, ARTIFACT_LOCATIONS.md, CLAUDE.md, PR_CHECKLIST.md, capitalization README
- [ ] In-flight exploratory experiments archived
- [ ] `claude.local.md.template` updated with `ck-projects/` path

## Out of scope

- Long-term FT template strategy — skills generate configs now; revisit if shared templates emerge as a need
- Spinning out `model_organisms` as a separate repo — explicitly not on the table
- Blueprint divergence policy (what happens when a scratch snapshot improves on the blueprint) — TBD in a follow-up
- Preservation of other sanity_checks' research lineage — they're being deleted; speak up before merge if one deserves to become a project

## Related work

- Wiki proposal: [File-re‐org-proposal](https://github.com/niznik-dev/cruijff_kit/wiki/File-re%E2%80%90org-proposal)
- Parent checklist: [Beta-Readiness](https://github.com/niznik-dev/cruijff_kit/wiki/Beta-Readiness)
- Related issues: #391 (prefer eval metadata over string-parsing paths), #382 (workflow test plaintext), #436 (multi-dataset re-add)

## Open for group input

- Preservation of any specific `sanity_checks/` lineage (all 5 non-model_organism members currently slated for deletion)
- Blueprint divergence policy (out-of-scope for this PR but worth sketching for the follow-up)

—MxC

### PR DESCRIPTION
**Pull request by niznik-dev**
Closes #437

# Description

Executes the atomic folder reorg committed in `edi-ccore/reviews/folder_reorg_plan_2026-04-22.md`. Deviations from the plan are documented in the addendum at the bottom of that same doc.

**What changed (15 commits, ~40% of tracked repo files touched):**

### `src/` layout (chunk 2)
- Library code moved under `src/` (flat layout): `src/tools/`, `src/utils/`, `src/tabular_to_text_gen/`
- `pyproject.toml`: `package-dir = {cruijff_kit = "src"}` — imports remain `from cruijff_kit.X import Y`
- `tabular_to_text_gen.*` imports rewritten to `cruijff_kit.tabular_to_text_gen.*` across tree

### `blueprints/` (chunks 4 + F)
- 5-slot shape (README, inspect_task.py, generate_data.py, optional modifiers/ + baseline.py)
- `blueprints/capitalization/` + `blueprints/folktexts/` built from `experiments/*/` content
- `blueprints/model_organism/`: README-only pointer; framework code stays in `src/tools/model_organisms/` (see addendum)
- `experiments/` directory retired entirely
- `synthetic_twins` retired — superseded by model_organisms framework
- **Chunk F:** initially landed as `projects/`, then renamed to `blueprints/` to disambiguate from the `ck-projects/{project}/` runtime scratch concept (group meeting decision, 2026-04-27). YAML schema field `experiment.project` stays as-is — the lookup pattern is now `blueprints/{experiment.project}/`.

### `src/tools/model_organisms/` (chunk 3)
- `sanity_checks/model_organisms/` → `src/tools/model_organisms/` (peer of torchtune, inspect, slurm, experiment)
- `sanity_checks/` umbrella retired (other 5 subdirs already deleted in #434)

### Unified `ck-projects/` scratch (chunk 5)
- `ck-experiments/` + `ck-outputs/` + `ck-sanity-checks/` → `ck-projects/{project}/{experiment_name}/`
- Configs, checkpoints, logs, eval outputs now nest inside each experiment dir per run
- `experiment.project` field added (REQUIRED); `experiment.type` / `metadata.type` removed
- `claude.local.md.template`, test fixtures, skill docs, workflow test specs all updated

### Parquet workflow support removed (chunk 1)
- Dropped from `setup_finetune.py` fine-tuning path (unused for months)
- Preserved in `tabular_to_text_gen` module as an input format (Sarah's skill)

### Housekeeping (chunk 1d)
- Deleted capitalization/misc, logs, package-lock.json, utils/ (unreferenced)
- `calc_token_stats.py` promoted to `src/utils/`
- `inspect_task_general.py` + design doc retired (orphaned 3 months, no automated callers)
- `experiments/folktexts/create_balanced_4223145.out` (187KB stray) removed

### Doc sweeps
- ARCHITECTURE.md, SKILLS_ARCHITECTURE_SUMMARY.md, WORKFLOW_GUIDE.md, ACS_EXAMPLE.md, SUPPORTED_MODELS.md, all skill docs, all agent specs updated for new paths and terminology
- Historical "tasks/ → experiments/ rename" narrative replaced with current-state terminology
- Post-reorg skill audit: 38 findings dispositioned across 11 follow-up commits (`edi-ccore/reviews/skill_audit_post_reorg_2026-04-27.md`)

## New Dependencies

None.

## Existing user data migration

Existing scratch in `ck-experiments/`, `ck-outputs/`, and `ck-sanity-checks/` stays put. New runs will land in `ck-projects/{project}/{experiment_name}/`. Users migrate their own historical runs at their leisure.

`ck-out-` prefix on run output dirs preserved for this PR (cosmetic debt — removing it would require `src/tools/torchtune/setup_finetune.py` logic changes; tracked as a follow-up).

# Testing Instructions

**Unit tests pass:** `pytest tests/unit/` → 906/906 (gained 2 new tests for `experiment.project` schema validation, plus skill-audit follow-ups). Run from repo root after `pip install -e .`.

**Workflow tests:** The 3 `.claude/workflow_test*.yaml` specs were updated and one end-to-end variant was run successfully on 2026-04-27 (model_organism `last_digits_k10`, single fine-tune Llama-3.2-1B-Instruct, both jobs completed with accuracy 1.0). The 3 standard variants haven't been re-run after the chunk F rename — quick sanity check suggested before merge:
- `test the workflow` (LoRA Comparison — rank4/rank8) is the most load-bearing
- Verifies scaffold → run pipeline generates `ck-projects/capitalization/workflow_test_{date}/` correctly using `blueprints/capitalization/inspect_task.py`
- Will catch any stale path refs I missed

**Manual smoke checks worth running before merge:**
1. `from cruijff_kit.tools.model_organisms.generate import generate` imports cleanly
2. `from cruijff_kit.tabular_to_text_gen.convert import main` imports cleanly
3. `python -m cruijff_kit.tools.model_organisms.generate --help` works
4. Reading any `blueprints/{capitalization,folktexts,model_organism}/README.md` reflects current state

—MxC

### CODE REVIEW

**Comment by niznik-dev** [comment]
### Unsettled: where should generated data live?

Surfaced during the post-reorg workflow test on this branch (`last_digits_k10_2026-04-27` model_organism experiment). Two different patterns are currently in use across the toolkit:

| Data type | Where it landed | Convention source |
|---|---|---|
| **model_organism** generated data (rule/k/N/seed-specific) | Inside the experiment dir (`ck-projects/{project}/{experiment_name}/dataset.json`) | `output_path` in the `data_generation` block resolves relative to experiment dir |
| **folktexts**, **capitalization** datasets | Centralized at `{ck_data_dir}/{project}/` outside any experiment | The data tier migration we just did in this PR |

There's a defensible principle behind each:

- **Co-locate (current model_organism behavior):** the experiment is self-contained — delete the experiment dir, you delete its data. Per-experiment generated synthetic data isn't reusable across runs anyway (different seeds/params). No risk of polluting a shared namespace.
- **Centralize (current folktexts/capitalization behavior):** real datasets are shared across many experiments. Generating into `ck-data/` once and pointing at it from many `experiment_summary.yaml`s avoids 10-100MB of duplication per experiment.

Probably the right answer is "both, depending on whether the data is per-experiment-specific or shared," documented somewhere — but worth a discussion before encoding either as the One True Way. Not blocking the merge of this PR; just flagging it as a follow-up.

—MxC

**Comment by msalganik** [comment]
@niznik-dev 

I'm not sure that we can/should distribute the folktexts and words data since it is not really ours. Another option is to just ask the user to get it themselves.  This is cleaner in some way. 

Happy to discuss on Friday or at next software meeting.

**Comment by niznik-dev** [comment]
A note to anyone reading this - never do this again. 197 files is too many. 😭
